app/jobs/sync_segments_job.rb

```diff
@@ -0,0 +1,22 @@
+class SyncSegmentsJob < ApplicationJob
+  include SyncSegmentsConverter
+
+  queue_as :default
+
+  def perform(*args)
```

Comment:
I wonder if these jobs could have been more parameterised rather than almost duplicating the read, process, log, notify code?